airbyte-integrations/connectors/source-metabase/source_metabase/components.py

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from dataclasses import dataclass
+from typing import Any, Mapping
+
+import requests
+from airbyte_cdk.sources.streams.http.auth import HttpAuthenticator
+from dataclasses_jsonschema import JsonSchemaMixin
+
+API_URL_CONFIG_KEY = "instance_api_url"
+USERNAME_CONFIG_KEY = "username"
+PASSWORD_CONFIG_KEY = "password"
+SESSION_TOKEN_CONFIG_KEY = "session_token"
+
+
+@dataclass
+class MetabaseAuth(HttpAuthenticator, JsonSchemaMixin):
+    def __init__(self, config: Mapping[str, Any], options: Mapping[str, Any]):
```

Comment:
One more thing: we used to have one session per sync, now we will have one session per stream. I suggest you make it a source instance attribute/cached property/etc